src/App.tsx

```diff
@@ -1,26 +1,342 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import {
+  USER_ID,
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import classNames from 'classnames';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
 
-const USER_ID = 0;
+export enum Status {
+  all = 'all',
+  active = 'active',
+  completed = 'completed',
+}
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState<Status>(Status.all);
+  const [id, setId] = useState<number>(1);
+
+  const [hasError, setHasError] = useState<boolean>(false);
+  // const [updateError, setUpdateError] = useState<boolean>(false);
+
+  const [errors, setErrors] = useState({
+    titleError: false,
+    todosError: false,
+    addError: false,
+    deleteError: false,
+    updateError: false,
+  });
+
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isSubmiting, setIsSubmiting] = useState<boolean>(false);
+  const [isDeleteing, setIsDeleteing] = useState<boolean>(false);
+  const [isUpdateing, setIsUpdateing] = useState<boolean>(false);
```

Comment:
loader should be on every todo, which updating or deleting.
this boolean states can't show loader for every todo which updating or deleting process.
suggest to make one state for todo whick in process on server. it can be array of todo ids with which waiting response from server...